dynamiqs/integrators/core/rouchon_integrator.py

```diff
@@ -48,82 +51,160 @@ def step(self, terms, t0, t1, y0, args, solver_state, made_jump):
     def func(self, terms, t0, y0, args):
         pass
 
-
-class Rouchon1DXSolver(RouchonDXSolver):
     def order(self, terms):
-        return 1
+        return self._order
 
 
-def cholesky_normalize(M0: QArray, LdL: QArray, dt: float, rho: QArray) -> jax.Array:
-    # To normalize the scheme, we compute S = M0d @ M0 + M1d @ M1 and replace
-    #   M0 by ~M0 = M0 @ S^{-1/2}
-    #   M1 by ~M1 = M1 @ S^{-1/2}
+def cholesky_normalize(Ms: list[QArray], rho: QArray) -> jax.Array:
+    # To normalize the scheme, we compute
+    #   S = sum_k Mk^† @ Mk
+    # and replace
+    #   Mk by ~Mk = Mk @ S^{-1/2}
```

Comment:
nit: I find this comment a bit indirect. I think it would be easier to follow to directly say that we use the cholesky decomposition S = T @ T^† instead of going through the square root, which is never used in the end. But it's not super important, it's just a comment.